From 810d0bf3ecd0e57683249a141f2cccf4b11eba13 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Wed, 6 Apr 2022 14:41:15 +0300
Subject: [PATCH 090/103] i2c: mxc: revert commit "s32: remove s32-gen1-regs
 header"

This reverts commit 3e26873cd1d3b8247dfc474dd3a76746e6613bcb with regard
to drivers/i2c/mxc_i2c changes.

Issue: ALB-8594
Upstream-Status: Pending 

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/i2c/mxc_i2c.c | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/drivers/i2c/mxc_i2c.c b/drivers/i2c/mxc_i2c.c
index b725d3f251..a792690f15 100644
--- a/drivers/i2c/mxc_i2c.c
+++ b/drivers/i2c/mxc_i2c.c
@@ -670,11 +670,21 @@ static int bus_i2c_write(struct mxc_i2c_bus *i2c_bus, u8 chip, u32 addr,
 	return ret;
 }
 
-#define I2C1_BASE_ADDR		(0x401E4000)
-#define I2C2_BASE_ADDR		(0x401E8000)
-#define I2C3_BASE_ADDR		(0x401EC000)
-#define I2C4_BASE_ADDR		(0x402D8000)
-#define I2C5_BASE_ADDR		(0x402DC000)
+#if !defined(I2C2_BASE_ADDR)
+#define I2C2_BASE_ADDR	0
+#endif
+
+#if !defined(I2C3_BASE_ADDR)
+#define I2C3_BASE_ADDR	0
+#endif
+
+#if !defined(I2C4_BASE_ADDR)
+#define I2C4_BASE_ADDR	0
+#endif
+
+#if !defined(I2C5_BASE_ADDR)
+#define I2C5_BASE_ADDR 0
+#endif
 
 #if !defined(I2C6_BASE_ADDR)
 #define I2C6_BASE_ADDR 0
-- 
2.25.1

